dart/lib/src/transport/rate_limiter.dart

```diff
@@ -22,6 +24,19 @@ class RateLimiter {
       if (_isRetryAfter(item.header.type)) {
         dropItems ??= [];
         dropItems.add(item);
+
+        final category = _categoryFromItemType(item.header.type);
+        if (category == DataCategory.transaction) {
+          _clientReportRecorder.recordLostEvent(
+            DiscardReason.rateLimitBackoff,
+            DataCategory.transaction,
+          );
+        } else {
+          _clientReportRecorder.recordLostEvent(
+            DiscardReason.rateLimitBackoff,
+            DataCategory.error,
+          );
```

Comment:
`h`: The rate limit could also apply to attachments or other categories. Now we would always record `.error` when the category is not a `.transaction`. Am I missing something?

Reply:
You are right, still need to do the mapping like in iOS with `SentryDataCategoryMapper`. 👍